src/firebolt/async_db/connection.py

```diff
@@ -158,6 +180,11 @@ async def _aclose(self) -> None:
         await self._client.aclose()
         self._is_closed = True
 
+    @cached_property
+    def _engine_name(self) -> str:
```

Comment:
Not a fan of having this field even with a private name. You think we could inline this?

Reply:
Fair point, this way there's no danger of using it in a place its failure would matter.